src/hdmf/utils.py

```diff
@@ -121,6 +125,51 @@ def __format_type(argtype):
         raise ValueError("argtype must be a type, str, list, or tuple")
 
 
+def __check_enum_argtype(argtype):
+    """Return True/False whether the given argtype is a supported docval enum type (bool, ints, floats, str)"""
+    if isinstance(argtype, (list, tuple)):
+        for i in argtype:
+            if not __check_enum_argtype(i):
+                return False
+        return True
+    return argtype in __supported_bool_types + __supported_int_types + __supported_float_types + (str, )
+
+
+def __check_enum(argval, arg):
+    """
+    Helper function to check whether the given argument value validates against the enum specification.
+
+    :param argval: argument value passed to the function/method
+    :param arg: argument validator - the specification dictionary for this argument
+
+    :return: None if the value validates successfully, error message if the value does not.
+    """
+    allowed_vals = arg['enum']
+    if not isinstance(allowed_vals, (list, tuple)):
+        msg = 'docval for %s: enum value must be a list or tuple (received %s)' % (arg['name'], type(allowed_vals))
```

Comment:
Is this checking that the specified enum is a valid enum?

Reply:
I think so, and I think maybe this should be in another function. Maybe the logical flow should be

1) validate individual docval args - that the specified enum is a valid enum
2) validate internal consistency of docval - that the types within the supplied enum match the allowed dtypes for that arg
3) validate the supplied data against the specified enum

Would it be better to have these steps be separated out into: 

1) docval validator (makes sure that the docval setup is valid)
2) arg validator (using the valid docval args)

Or is it better to have each of these 3 steps for each check?

Reply:
Yeah, I think it makes sense to have a separate docval validator to check for internal consistency and an arg validator. I think the docval validator should be done within the `dec` function on line 497, so it is not called when the function is called but rather when the module is loaded. I will refactor that.

Reply:
The flow is now:
When the module is loaded, docval internal check:
1. check that enum values is a list or tuple
2. check that argument type is compatible with use of enum (if type is allowed to be TimeSeries, then enum is not allowed)
3. check that each enum value is of a type listed in the argument types

When the decorated function is called, check that the supplied data is one of the values listed in the enum values.